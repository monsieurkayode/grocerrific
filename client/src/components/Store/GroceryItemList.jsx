import React from 'react';

import GroceryItem from './GroceryItem';

const GroceryItemList = () => (
  <section id="grocery__display">
    <div className="grocery__list">
      <GroceryItem />
      <GroceryItem />
      <GroceryItem />
      <GroceryItem />
      <GroceryItem />
      <GroceryItem />
      <GroceryItem />
      <GroceryItem />
    </div>
  </section>
);

export default GroceryItemList;
