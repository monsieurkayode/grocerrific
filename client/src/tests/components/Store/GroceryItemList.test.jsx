import React from 'react';
import { shallow } from 'enzyme';
import GroceryItemList from '../../../components/Store/GroceryItemList';
import mockCartItems from '../../__mocks__/mockCartItems';
import mockGroceries from '../../__mocks__/mockGroceries';

const setup = () => {
  const props = {
    cartItems: mockCartItems,
    groceries: mockGroceries,
    addToCart: jest.fn(),
    removeFromCart: jest.fn(),
    loading: false
  };

  return shallow(<GroceryItemList {...props} />);
};

describe('GroceryItemList Component', () => {
  const wrapper = setup();
  afterEach(() => wrapper.setProps({ loading: false }));

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a Loader when loading', () => {
    wrapper.setProps({ loading: true });
    expect(wrapper.find('Loader').length).toBe(1);
  });

  it('should render three GroceryItem children component', () => {
    expect(wrapper.find('GroceryItem').length).toBe(3);
  });

  it('should render a NoContent component if there are no groceries', () => {
    wrapper.setProps({ groceries: [] });
    expect(wrapper.find('NoContent').length).toBe(1);
  });
});
