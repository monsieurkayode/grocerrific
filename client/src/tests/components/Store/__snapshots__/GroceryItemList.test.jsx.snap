// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroceryItemList Component renders correctly 1`] = `
<section
  id="grocery__display"
>
  <div
    className="grocery__list"
  >
    <GroceryItem
      _id="5bf4754223602e00211d6587"
      addToCart={[MockFunction]}
      cartItems={
        Array [
          Object {
            "id": "5bf4754223602e00211d6587",
            "maxQuantity": 4,
            "name": "mamador",
            "price": 1250,
            "quantity": 3,
          },
          Object {
            "id": "5bf4754223602e00211d6588",
            "maxQuantity": 15,
            "name": "sausage",
            "price": 120,
            "quantity": 12,
          },
          Object {
            "id": "5bf4754223602e00211d6589",
            "maxQuantity": 10,
            "name": "salad creme",
            "price": 1500,
            "quantity": 1,
          },
        ]
      }
      key="5bf4754223602e00211d6587"
      name="mamador"
      price={1250}
      quantity={3}
      removeFromCart={[MockFunction]}
    />
    <GroceryItem
      _id="5bf4754223602e00211d6588"
      addToCart={[MockFunction]}
      cartItems={
        Array [
          Object {
            "id": "5bf4754223602e00211d6587",
            "maxQuantity": 4,
            "name": "mamador",
            "price": 1250,
            "quantity": 3,
          },
          Object {
            "id": "5bf4754223602e00211d6588",
            "maxQuantity": 15,
            "name": "sausage",
            "price": 120,
            "quantity": 12,
          },
          Object {
            "id": "5bf4754223602e00211d6589",
            "maxQuantity": 10,
            "name": "salad creme",
            "price": 1500,
            "quantity": 1,
          },
        ]
      }
      key="5bf4754223602e00211d6588"
      name="sausage"
      price={120}
      quantity={12}
      removeFromCart={[MockFunction]}
    />
    <GroceryItem
      _id="5bf4754223602e00211d6589"
      addToCart={[MockFunction]}
      cartItems={
        Array [
          Object {
            "id": "5bf4754223602e00211d6587",
            "maxQuantity": 4,
            "name": "mamador",
            "price": 1250,
            "quantity": 3,
          },
          Object {
            "id": "5bf4754223602e00211d6588",
            "maxQuantity": 15,
            "name": "sausage",
            "price": 120,
            "quantity": 12,
          },
          Object {
            "id": "5bf4754223602e00211d6589",
            "maxQuantity": 10,
            "name": "salad creme",
            "price": 1500,
            "quantity": 1,
          },
        ]
      }
      key="5bf4754223602e00211d6589"
      name="salad creme"
      price={1500}
      quantity={0}
      removeFromCart={[MockFunction]}
    />
  </div>
</section>
`;
